iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/scheduler/load/LoadTsFileScheduler.java

```diff
@@ -143,69 +143,73 @@ public LoadTsFileScheduler(
-              secondPhase(isFirstPhaseSuccess, uuid, node.getTsFileResource());
-
-          node.clean();
-          if (!isFirstPhaseSuccess || !isSecondPhaseSuccess) {
-            isLoadSingleTsFileSuccess = false;
+    try {
+      stateMachine.transitionToRunning();
+      int tsFileNodeListSize = tsFileNodeList.size();
+      boolean isLoadSuccess = true;
+
+      for (int i = 0; i < tsFileNodeListSize; ++i) {
+        LoadSingleTsFileNode node = tsFileNodeList.get(i);
+        boolean isLoadSingleTsFileSuccess = true;
+        try {
+          if (node.isTsFileEmpty()) {
+            LOGGER.info(
+                "Load skip TsFile {}, because it has no data.",
+                node.getTsFileResource().getTsFilePath());
+
+          } else if (!node.needDecodeTsFile(
+              slotList ->
+                  partitionFetcher.queryDataPartition(
+                      slotList,
+                      queryContext.getSession().getUserName()))) { // do not decode, load locally
+            isLoadSingleTsFileSuccess = loadLocally(node);
+            node.clean();
+
+          } else { // need decode, load locally or remotely, use two phases method
+            String uuid = UUID.randomUUID().toString();
+            dispatcher.setUuid(uuid);
+            allReplicaSets.clear();
+
+            boolean isFirstPhaseSuccess = firstPhase(node);
+            boolean isSecondPhaseSuccess =
+                secondPhase(isFirstPhaseSuccess, uuid, node.getTsFileResource());
+
+            node.clean();
+            if (!isFirstPhaseSuccess || !isSecondPhaseSuccess) {
+              isLoadSingleTsFileSuccess = false;
+            }
           }
-        }
-        if (isLoadSingleTsFileSuccess) {
-          LOGGER.info(
-              "Load TsFile {} Successfully, load process [{}/{}]",
-              node.getTsFileResource().getTsFilePath(),
-              i + 1,
-              tsFileNodeListSize);
-        } else {
+          if (isLoadSingleTsFileSuccess) {
+            LOGGER.info(
+                "Load TsFile {} Successfully, load process [{}/{}]",
+                node.getTsFileResource().getTsFilePath(),
+                i + 1,
+                tsFileNodeListSize);
+          } else {
+            isLoadSuccess = false;
+            LOGGER.warn(
+                "Can not Load TsFile {}, load process [{}/{}]",
+                node.getTsFileResource().getTsFilePath(),
+                i + 1,
+                tsFileNodeListSize);
+          }
+        } catch (Exception e) {
           isLoadSuccess = false;
+          stateMachine.transitionToFailed(e);
           LOGGER.warn(
-              "Can not Load TsFile {}, load process [{}/{}]",
-              node.getTsFileResource().getTsFilePath(),
-              i + 1,
-              tsFileNodeListSize);
+              String.format(
+                  "LoadTsFileScheduler loads TsFile %s error",
+                  node.getTsFileResource().getTsFilePath()),
+              e);
```

Comment:
```suggestion
              "LoadTsFileScheduler loads TsFile {} error",
              node.getTsFileResource().getTsFilePath(),
              e);
```

Reply:
Thank you for reviewing! I will modify my code according to your elegant comments!